firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -551,8 +563,24 @@ class BrowserCoordinator: BaseCoordinator,
         browserViewController.updateZoomPageBarVisibility(visible: true)
     }
 
-    func showShareSheet(with url: URL?) {
-        showShareSheet(with: url, title: nil)
+    /// Share the currently selected tab using the share sheet.
+    ///
+    /// Part of the MainMenuCoordinatorDelegate implementation, called from the New Menu > Tools > Share.
+    func showShareSheetForCurrentlySelectedTab() {
+        // We share the tab's displayURL to make sure we don't share reader mode localhost URLs
+        guard let selectedTab = tabManager.selectedTab,
+              let url = selectedTab.canonicalURL?.displayURL else {
+            return
+        }
```

Comment:
I am wondering if the coordinator should be responsible of passing down this information. Not sure where in the chain this would go otherwise, just wanted to ask your opinion about it